compiler/linkname.go

```diff
@@ -98,63 +166,37 @@ func parseGoLinknames(fset *token.FileSet, pkgPath string, file *ast.File) ([]Go
 	isUnsafe := astutil.ImportsUnsafe(file)
 
 	processComment := func(comment *ast.Comment) error {
-		if !strings.HasPrefix(comment.Text, "//go:linkname ") {
-			return nil // Not a linkname compiler directive.
+		link, err := readLinknameFromComment(pkgPath, comment)
+		if err != nil || link == nil {
+			return err
 		}
 
-		// TODO(nevkontakte): Ideally we should check that the directive comment
-		// is on a line by itself, line Go compiler does, but ast.Comment doesn't
-		// provide an easy way to find that out.
```

Comment:
@nevkontakte Please double check that the `if len(fields) != 3 {` does this. It looks like it was already checking that there wasn't anything after the `extname`, so I'm assuming this has already been managed. However, I don't want to remove this `TODO` if I'm just misunderstanding. I did add a unit-test to check for stuff after the `extname`.

Reply:
I don't remember exactly what I meant here 🤔 It may have been related to multi-line comments and how compiler directive must be separated in them?.. But since the meaning of the TODO is uncertain, it's value is also questionable, so I don't mind dropping it either way.